tpr/tpr.go

```diff
@@ -0,0 +1,120 @@
+package tpr
+
+import (
+	"fmt"
+	"time"
+
+	microerror "github.com/giantswarm/microkit/error"
+	"github.com/rossf7/pingdom-operator/pkg/util"
```

Comment:
We can use https://godoc.org/github.com/cenkalti/backoff#ExponentialBackOff explicitly if it makes sense. Lets not reinvent the wheel and just use what the community provides. I would also like to make the backoff configurable, then clients can simply provide one with the configuration they favour and we only provide a default one we think is right.